### PR TITLE
Fix the scripts after the review session

### PR DESCRIPTION
**Pull request by nhantran2642**

### CODE REVIEW

File: src/main/java/com/nhantran/pages/BasePage.java [hunk 1]
```diff
@@ -8,18 +9,19 @@ public class BasePage {
 
     private String railwayTab = "//div[@id='menu']//li/a/span[text()='%s']";
```
Comment (vihoang0706) [inline src/main/java/com/nhantran/pages/BasePage.java:10]: rename selector variable to easier distigush, it's element (e/x: dynamicMenuTabSelector,...)

File: src/main/java/com/nhantran/pages/BookTicketPage.java [hunk 2]
```diff
@@ -1,41 +1,80 @@
...
+            EnumSet.of(BookTicketComboBoxes.DEPART_STATION, BookTicketComboBoxes.ARRIVE_STATION);
 
+    public String getStation(BookTicketComboBoxes comboBoxName) {
+        if (!ALLOWED_STATIONS.contains(comboBoxName)) {
+            throw new IllegalArgumentException("Invalid combobox: " + comboBoxName);
+        }
+        return SeleniumActions.getSelectedOption(By.xpath(String.format(selectBox, comboBoxName.getValue())));
+    }
+
+    public String calculateNextDate(int numberOfNextDays) {
```
Comment (vihoang0706) [inline src/main/java/com/nhantran/pages/BookTicketPage.java:75]: it does not belong to this page. it's better if put it under utils

File: src/main/java/com/nhantran/pages/ForgetPasswordPage.java [hunk 3]
```diff
@@ -4,12 +4,12 @@
 import org.openqa.selenium.By;
 
 public class ForgetPasswordPage extends BasePage {
-    By txtEmail = By.xpath("//input[@id='email']");
-    By sendInstructionButton = By.xpath("//input[@type='submit']");
+    private By txtEmail = By.xpath("//input[@id='email']");
```
Comment (vihoang0706) [inline src/main/java/com/nhantran/pages/ForgetPasswordPage.java:7]: Should be consistent when defining variables, base their names on the type of element (textbox, combobox, button, label, combobox,...):
1. Use abbreviated names: txtEmail, btnSendInstructions, lblSuccessMessage, etc.
2. Use full names indicating the type of element: emailTextBox, sendInstructionsButton, etc.
Ensure consistency across pages

File: src/main/java/com/nhantran/pages/ForgetPasswordPage.java [hunk 4]
```diff
@@ -4,12 +4,12 @@
...
+    private By sendInstructionButton = By.xpath("//input[@type='submit']");
 
 
     public void getResetPasswordLink(String email) {
```
Comment (vihoang0706) [inline src/main/java/com/nhantran/pages/ForgetPasswordPage.java:11]: rename function, normally get often returns a value

File: src/main/java/com/nhantran/pages/TimetablePage.java [hunk 5]
```diff
@@ -1,13 +1,21 @@
...
 public class TimetablePage extends BasePage {
 
-    private String checkPriceLink = "//tr[td[text()='%s' and following-sibling::td[text()='%s']]]//a[text()='check price']";
+    private String lnkCheckPrice = "//tr[td[text()='%s' and following-sibling::td[text()='%s']]]//a[text()='check price']";
+    private String linkBookTicket = "//tr[td[text()='%s' and following-sibling::td[text()='%s']]]//a[text()='book ticket']";
 
-    public void clickCheckPriceLink(String departStation, String arrivalStation) {
-        SeleniumActions.clickElement((By.xpath(String.format(checkPriceLink, departStation, arrivalStation))));
+    public void clickCheckPriceLink(RailwayStations departStation, RailwayStations arrivalStation) {
+        SeleniumActions.scrollToElement((By.xpath(String.format(lnkCheckPrice, departStation.getValue(), arrivalStation.getValue()))));
```
Comment (vihoang0706) [inline src/main/java/com/nhantran/pages/TimetablePage.java:13]: create a variable to assign for this: String.format(lnkCheckPrice, departStation.getValue(), arrivalStation.getValue()

=> reduce duplicate

File: src/main/java/com/nhantran/pages/TimetablePage.java [hunk 6]
```diff
@@ -1,13 +1,21 @@
...
+        SeleniumActions.clickElement((By.xpath(String.format(lnkCheckPrice, departStation.getValue(), arrivalStation.getValue()))));
+    }
+
+    public void clickBookTicketLink(RailwayStations departStation, RailwayStations arrivalStation) {
+        SeleniumActions.scrollToElement((By.xpath(String.format(linkBookTicket, departStation.getValue(), arrivalStation.getValue()))));
```
Comment (vihoang0706) [inline src/main/java/com/nhantran/pages/TimetablePage.java:18]: same above

File: src/main/java/com/nhantran/pages/RegistrationConfirmationPage.java [hunk 7]
```diff
@@ -1,17 +1,14 @@
...
 import org.openqa.selenium.By;
 
-import java.time.Duration;
-import java.util.concurrent.TimeUnit;
 
-public class RegistrationConfirmationPage extends BasePage{
+public class RegistrationConfirmationPage extends BasePage {
 
-    By confirmSuccessMessage = By.xpath("//p");
+    By lblConfirmSuccess = By.xpath("//p");
```
Comment (vihoang0706) [inline src/main/java/com/nhantran/pages/RegistrationConfirmationPage.java:9]: add access modifier

File: src/test/java/com/nhantran/base/TestBase.java [hunk 8]
```diff
@@ -15,24 +16,14 @@
...
-    protected static String invalidPassword;
-    protected static String notActiveUsername;
-    protected static String notActivePassword;
+    private Properties properties;
 
     public TestBase() {
         try {
             properties = new Properties();
             FileInputStream fis = new FileInputStream(
                     System.getProperty("user.dir") + "/src/main/java/com/nhantran/config/config.properties");
```
Comment (vihoang0706) [inline src/test/java/com/nhantran/base/TestBase.java:25]: should put these paths under constants

File: src/test/java/com/nhantran/base/TestBase.java [hunk 9]
```diff
@@ -41,17 +32,17 @@ public TestBase() {
         }
     }
 
-    protected static void startAUT() {
+    protected void startAUT() {
         String browser = properties.getProperty("browser");
-        String url = properties.getProperty("url");
+        String url = Constants.RAILWAY_URL;
```
Comment (vihoang0706) [inline src/test/java/com/nhantran/base/TestBase.java:37]: railway url is defining 2 places: config.properties and constants????

File: src/test/java/com/nhantran/chapter10/BookTicketTest.java [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+        Assert.assertEquals(bookTicketSuccessPage.getTicketAmount(), ticket.getTicketAmount(), "Ticket amount not matching");
+    }
+
+    @Test(description = "User can check price of ticket from Timetable")
+    public void TC014_SuccessToCheckTicketPriceFromTimetable() {
+        homePage.clickTab(RailwayTabs.LOGIN);
+        loginPage.login(validUser);
+        homePage.clickTab(RailwayTabs.TIMETABLE);
+        timetablePage.clickCheckPriceLink(RailwayStations.DA_NANG, RailwayStations.SAI_GON);
+        Assert.assertEquals(ticketPricePage.isPageTitleDisplayed("Ticket Price"), Boolean.TRUE, "Page does not load");
```
Comment (vihoang0706) [inline src/test/java/com/nhantran/chapter10/BookTicketTest.java:55]: research for assertTrue

File: src/test/java/com/nhantran/chapter10/LoginTest.java [hunk 11]
```diff
@@ -1,63 +1,58 @@
...
-    private String loginContentErrorMessage = "There was a problem with your login and/or errors exist in your form.";
-    private String loginAccountErrorMessage = "Invalid username or password. Please try again.";
+    private HomePage homePage = new HomePage();
+    private LoginPage loginPage = new LoginPage();
+    private User validUser = new User(Constants.VALID_USERNAME, Constants.VALID_PASSWORD);
+    private User invalidUser = new User(Constants.VALID_USERNAME, Constants.INVALID_PASSWORD);
+    private User nonActiveUser = new User(Constants.NON_ACTIVE_USERNAME, Constants.NON_ACTIVE_PASSWORD);
 
     @Test(description = "User can log into Railway with valid username and password")
     public void TC001_SuccessToLoginWithValidAccount() {
```
Comment (vihoang0706) [inline src/test/java/com/nhantran/chapter10/LoginTest.java:22]: the method's name should start with verb (e/x: TC001_LoginWithValidAccount, TC001_AbleToLoginWithValidAccount,....)

File: src/test/java/com/nhantran/chapter10/LoginTest.java [hunk 12]
```diff
@@ -1,63 +1,58 @@
...
-    public LoginTest() {
-        super();
-    }
-
-    private String loginContentErrorMessage = "There was a problem with your login and/or errors exist in your form.";
-    private String loginAccountErrorMessage = "Invalid username or password. Please try again.";
+    private HomePage homePage = new HomePage();
+    private LoginPage loginPage = new LoginPage();
+    private User validUser = new User(Constants.VALID_USERNAME, Constants.VALID_PASSWORD);
+    private User invalidUser = new User(Constants.VALID_USERNAME, Constants.INVALID_PASSWORD);
```
Comment (vihoang0706) [inline src/test/java/com/nhantran/chapter10/LoginTest.java:18]: inactive

File: src/test/java/com/nhantran/chapter10/CreateAccountTest.java [hunk 13]
```diff
@@ -1,61 +1,58 @@
...
+    private User emptyPasswordPidUser = new User(Constants.VALID_USERNAME, null, null, null);
+    private User newUser;
 
     @Test(description = "User can't create account with an already in-use email")
     public void TC007_FailToCreateAccountWithEmailAlreadyUsed() {
-        homePage.clickTab("Register");
-        registerPage.register(validUsername, validPassword, validPassword, "0909090909090909");
+        homePage.clickTab(RailwayTabs.REGISTER);
+        registerPage.register(alreadyRegistedUser);
         Assert.assertEquals(registerPage.getErrorMessageAboveRegisterForm(), "This email address is already in use.", "The error message does not match");
```
Comment (vihoang0706) [inline src/test/java/com/nhantran/chapter10/CreateAccountTest.java:29]: should store these error messages into another file (e/x: constants) to easy to manage

File: src/test/java/com/nhantran/base/TestBase.java (hunk 9 above)
Comment (nhantran2642) [reply src/test/java/com/nhantran/base/TestBase.java:37]: in the previous PR, I stored the railway url in the config.properties, but after the review, I stored it in the Constants.

**Pull request closed without merging**